stdlib/SparseArrays/src/linalg.jl

```diff
@@ -7,6 +7,18 @@ using Random: rand!
 const StridedOrTriangularMatrix{T} = Union{StridedMatrix{T}, LowerTriangular{T}, UnitLowerTriangular{T}, UpperTriangular{T}, UnitUpperTriangular{T}}
 const AdjOrTransStridedOrTriangularMatrix{T} = Union{StridedOrTriangularMatrix{T},Adjoint{<:Any,<:StridedOrTriangularMatrix{T}},Transpose{<:Any,<:StridedOrTriangularMatrix{T}}}
 
++(A::Hermitian, B::SparseMatrixCSC) = parent(A) + B
++(A::SparseMatrixCSC, B::Hermitian) = A + parent(B)
+function +(A::Hermitian{<:Any, <:SparseMatrixCSC}, B::Hermitian{<:Any, <:SparseMatrixCSC})
+    Hermitian(parent(A) + parent(B))
```

Comment:
You were rightly concerned about the triangle issue. So the `parent`s don't need to be Hermitian themselves, plus the values that you care about maybe in different triangles for `A` and `B`. In fact, with the current implementation, they even may have the same `:L` triangle, yet the `Hermitian` constructor assumes `:U` by default. So the result could be entirely meaningless.

I think taking care of the orientation can be done by along the lines
```julia
if A.uplo == B.uplo
    return Hermitian(parent(A) + parent(B), sym_uplo(A.uplo))
elseif A.uplo == 'U'
    return Hermitian(parent(A) + permutedims(parent(B)), :U)
else
    return Hermitian(permutedims(parent(A)) + parent(B), :U)
end
```
or so...

Reply:
Seems like you need to `import LinearAlgebra: sym_uplo`.